File: FestivalC#/Festival/Utils/AbstractObservable.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Utils
{
    public class AbstractObservable<E> : Observable<E>
    {
        private List<Observer<E>> observers = new List<Observer<E>>();

        public void addObserver(Observer<E> o)
        {
            observers.Add(o);
        }

        public void notifyPushObservers(E e)
        {
            foreach (Observer<E> o in observers)
            {
                o.pushUpdate(e);
            }
        }

        public void removeObserver(Observer<E> o)
        {
            observers.Remove(o);
        }
    }
}

File: FestivalJava/Utils/src/main/java/ObserverPattern/AbstractObservable.java
package ObserverPattern;

import java.util.ArrayList;
import java.util.List;

/**
 * Created by Sebi on 30-Mar-17.
 */
public abstract class AbstractObservable<E> implements Observable<E> {
    private List<Observer<E>> observers = new ArrayList<Observer<E>>();

    @Override
    public void addObserver(Observer<E> o) {
        observers.add(o);
    }

    @Override
    public void removeObserver(Observer<E> o) {
        observers.remove(o);
    }

    @Override
    public void notifyObservers(){
        for (Observer<E> observer : observers){
            observer.update();
        }
    }
}
